File: server/src/main/scala/kpn/api/common/location/LocationNodesPage.scala
package kpn.api.common.location

import kpn.api.common.TimeInfo

case class LocationNodesPage(
  timeInfo: TimeInfo,
  summary: LocationSummary,
  nodeCount: Long,
  allNodeCount: Long,
  integrityCheckFailedNodeCount: Long,
  filter: LocationNodeOptions,
  nodes: Seq[LocationNodeInfo]
)

File: backend/api/src/main/java/kpn/api/common/location/LocationNodesPage.java
package kpn.api.common.location;

import kpn.api.common.TimeInfo;
import kpn.api.common.location.LocationNodeInfo;
import kpn.api.common.location.LocationNodeOptions;
import kpn.api.common.location.LocationSummary;

import com.google.common.collect.ImmutableList;

public record LocationNodesPage(
  TimeInfo timeInfo,
  LocationSummary summary,
  Long nodeCount,
  LocationNodeOptions filter,
  ImmutableList<LocationNodeInfo> nodes
) {
}
